weld/ast.rs

```diff
@@ -111,20 +111,79 @@ impl Type {
 pub enum ScalarKind {
     Bool,
     I8,
+    I16,
     I32,
     I64,
+    U8,
     F32,
     F64,
 }
 
+impl ScalarKind {
+    /// Is this scalar a floating point type (F32 or F64)?
+    pub fn is_float(&self) -> bool {
+        use ast::ScalarKind::*;
+        match *self {
+            F32 | F64 => true,
+            _ => false
+        }
+    }
+
+    /// Is this scalar a Bool?
+    pub fn is_bool(&self) -> bool {
+        use ast::ScalarKind::*;
+        match *self {
+            Bool => true,
+            _ => false
+        }
+    }
+
+    /// Is this type a signed integer of any type (excludes Bool)?
+    pub fn is_signed_integer(&self) -> bool {
+        use ast::ScalarKind::*;
+        match *self {
+            I8 | I16 | I32 | I64 => true,
+            _ => false
+        }
+    }
+
+    /// Is this type a signed integer of any type (excludes Bool)?
+    pub fn is_unsigned_integer(&self) -> bool {
+        use ast::ScalarKind::*;
+        match *self {
+            U8 => true,
```

Comment:
Should we also add `u16`, `u32`, and `u64` while we're at it?

Reply:
Good point, might as well add them